homeassistant/components/mpris/media_player.py

```diff
@@ -0,0 +1,640 @@
+        return self._integration_id + "-" + self.coordinator_context
+
+    @property
+    def name(self) -> str:
+        """Return the name of the entity."""
+        return self.coordinator_context
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return the device information associated with the entity."""
+        return DeviceInfo(
+            identifiers={(DOMAIN, self._integration_id)},
+            name=f"MPRIS agent at {self.coordinator.client.host}",
+            manufacturer="Freedesktop",
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """Entity has been added to HASS."""
+        await super().async_added_to_hass()
+        # And now update all the properties based on the info we have.
+        self._handle_coordinator_update()
+
+    async def async_media_play(self) -> None:
+        """Begin playback."""
+        try:
+            await self.coordinator.client.play(self.coordinator_context)
+        except Exception as exc:
+            raise HomeAssistantError("cannot play: %s" % exc) from exc
+
+    async def async_media_pause(self) -> None:
+        """Pause playback."""
+        try:
+            await self.coordinator.client.pause(self.coordinator_context)
+        except Exception as exc:
+            raise HomeAssistantError("cannot pause: %s" % exc) from exc
+
+    async def async_media_stop(self) -> None:
+        """Stop playback."""
+        try:
+            await self.coordinator.client.stop(self.coordinator_context)
+        except Exception as exc:
+            raise HomeAssistantError("cannot stop: %s" % exc) from exc
+
+    async def async_media_next_track(self) -> None:
+        """Skip to next track."""
+        try:
+            await self.coordinator.client.next(self.coordinator_context)
+        except Exception as exc:
+            raise HomeAssistantError("cannot next: %s" % exc) from exc
+
+    async def async_media_previous_track(self) -> None:
+        """Skip to previous track."""
+        try:
+            await self.coordinator.client.previous(self.coordinator_context)
+        except Exception as exc:
+            raise HomeAssistantError("cannot previous: %s" % exc) from exc
+
+    async def async_media_seek(self, position: float) -> None:
+        """Send seek command."""
+        try:
+            trackid = self._metadata.get("mpris:trackid", None)
+            await self.coordinator.client.set_position(
+                self.coordinator_context,
+                trackid,
+                position,
+            )
+        except Exception as exc:
+            raise HomeAssistantError("cannot seek: %s" % exc) from exc
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Return entity specific state attributes."""
+        if self.state == MediaPlayerState.OFF:
+            return {}
+        return {ATTR_PLAYBACK_RATE: self._attr_playback_rate}
+
+
+class MPRISCoordinator(
+    DataUpdateCoordinator[
```

Comment:
Maybe I am misremembering here, but I was explicitly requested in prior reviews to move the custom idiosyncratic logic to a `DataUpdateCoordinator` (which makes sense to me).  The base class is only a protocol that has a single function `async_add_listener`.

We do get the benefit of `async_set_updated_data(...)` from the coordinator which is great because all the entities are pinged when an update comes in.

I'm keeping the coordinator dependency.  In fact, I am increasing my reliance on it, by simply using `self.async_set_updated_error` which is awesome.  The coordinator is well thought out and it handles quite a few corner cases I really do not feel like maintaining myself.